collect_app/src/test/java/org/odk/collect/android/preferences/AppConfigurationGeneratorTest.kt

```diff
@@ -1,32 +1,51 @@
 package org.odk.collect.android.preferences
 
-import androidx.test.core.app.ApplicationProvider
 import androidx.test.ext.junit.runners.AndroidJUnit4
 import org.hamcrest.MatcherAssert.assertThat
 import org.hamcrest.Matchers.`is`
 import org.json.JSONObject
 import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
-import org.odk.collect.android.TestSettingsProvider
-import org.odk.collect.android.application.Collect
-import org.odk.collect.android.configure.qr.JsonPreferencesGenerator
-import org.odk.collect.android.injection.DaggerUtils
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.odk.collect.android.configure.qr.AppConfigurationGenerator
+import org.odk.collect.android.configure.qr.AppConfigurationKeys
 import org.odk.collect.android.preferences.keys.AdminKeys
 import org.odk.collect.android.preferences.keys.GeneralKeys
+import org.odk.collect.android.preferences.source.SettingsProvider
+import org.odk.collect.android.projects.CurrentProjectProvider
 import org.odk.collect.projects.Project
+import org.odk.collect.testshared.InMemSettings
 
 @RunWith(AndroidJUnit4::class)
-class JsonPreferencesGeneratorTest {
-    private val settingsProvider = TestSettingsProvider.getSettingsProvider()
-    private lateinit var jsonPreferencesGenerator: JsonPreferencesGenerator
+class AppConfigurationGeneratorTest {
+    private val generalSettingsForProject1 = InMemSettings()
+    private val adminSettingsForProject1 = InMemSettings()
+
+    private val settingsProvider = mock<SettingsProvider> {
+        on { getGeneralSettings() } doReturn generalSettingsForProject1
+        on { getAdminSettings() } doReturn adminSettingsForProject1
+    }
+
+    private val currentProjectProvider: CurrentProjectProvider = mock {
+        on { getCurrentProject() } doReturn Project.Saved("1", "Project X", "X", "#cccccc")
+    }
+
+    val projectDetails = mapOf(
+        AppConfigurationKeys.PROJECT_NAME to "Project X",
+        AppConfigurationKeys.PROJECT_ICON to "X",
+        AppConfigurationKeys.PROJECT_COLOR to "#cccccc"
+    )
+
+    private lateinit var appConfigurationGenerator: AppConfigurationGenerator
 
     @Before
     fun setup() {
         settingsProvider.getGeneralSettings().clear()
```

Comment:
I don't think you'll need these `clear()` calls here.

Reply:
Fixed.